src/github.com/t3chguy/riot-static/riot-static.go

```diff
@@ -1,232 +0,0 @@
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package main
-
-import (
-	"fmt"
-	"github.com/gin-gonic/gin"
-	"github.com/t3chguy/riot-static/matrix-client"
-	"github.com/t3chguy/riot-static/utils"
-	"net/http"
-	"os"
-	"strconv"
-	"time"
-)
-
-// TODO Cache memberList+serverList until it changes
-
-const PublicRoomsPageSize = 20
-const RoomTimelineSize = 20
-const RoomMembersPageSize = 20
-
-func LoadPublicRooms(first bool) {
-	fmt.Println("Loading publicRooms")
-	resp, err := client.PublicRooms(0, "", "")
-
-	if err != nil {
-		// Only panic if first one fails, after that we only have outdated data (less important)
-		if first {
-			panic(err)
-		} else {
-			fmt.Println(err)
-		}
-	}
-
-	// Preallocate the maximum capacity possibly needed (if all rooms were world readable)
-	worldReadableRooms := make([]*matrix_client.Room, 0, len(resp.Chunk))
-
-	// filter on actually WorldReadable publicRooms
-	for _, x := range resp.Chunk {
-		if !x.WorldReadable {
-			continue
-		}
-
-		var room *matrix_client.Room
-		if existingRoom := client.GetRoom(x.RoomId); existingRoom != nil {
-			room = existingRoom
-		} else {
-			room = client.NewRoom(x)
-		}
-
-		// Append world readable r to the filtered list.
-		worldReadableRooms = append(worldReadableRooms, room)
-	}
-	client.SetRoomList(worldReadableRooms)
-}
-
-var client *matrix_client.Client
-
-func main() {
-	client = matrix_client.NewClient()
-
-	router := gin.Default()
-	router.SetHTMLTemplate(tpl)
-	router.Static("/assets", "./assets")
-
-	router.GET("/", func(c *gin.Context) {
-		page, skip, end := utils.CalcPaginationPage(c.DefaultQuery("page", "1"), PublicRoomsPageSize)
-		c.HTML(http.StatusOK, "rooms.html", gin.H{
-			"Rooms": client.GetRoomList(skip, end),
-			"Page":  page,
-		})
-	})
-
-	roomRouter := router.Group("/room/")
-	{
-		// Load room into request object so that we can do any clean up etc here
-		roomRouter.Use(func(c *gin.Context) {
-			roomID := c.Param("roomID")
-
-			if room := client.GetRoom(roomID); room != nil {
-				if room.LazyInitialSync() {
-					c.Set("Room", room)
-					c.Next()
-				} else {
-					c.HTML(http.StatusInternalServerError, "room_error.html", gin.H{
-						"Error": "Failed to load room.",
-						"Room":  room,
-					})
-					c.Abort()
-				}
-			} else {
-				c.String(http.StatusNotFound, "Room Not Found")
-				c.Abort()
-			}
-		})
-
-		roomRouter.GET("/:roomID/", func(c *gin.Context) {
-			c.Redirect(http.StatusTemporaryRedirect, "chat")
-		})
-
-		roomRouter.GET("/:roomID/chat", func(c *gin.Context) {
-			room := c.MustGet("Room").(*matrix_client.Room)
```

Comment:
Go typically uses `camelCase` and not `snake_case` in scenarios where you need to use multiple words for identifiers. They discourage the use of multiple words, but that's sometimes unavoidable. You could probably abbreviate this to `mxclient` or even `client` to be honest (I don't think there are any other clients this project will be dealing with, bearing in mind the HTTP client is `http.Client`).

https://golang.org/doc/effective_go.html#mixed-caps